### PR TITLE
docs: creating stacks

### PR DESCRIPTION
**Pull request by mhamwala**
initial PR for documentation on creating stacks

### CODE REVIEW

File: docs/stacks/create-or-modify.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+git clone https://github.com/appsody/docs
+cd docs/sample-stack
+```
+
+## Modifying a stack
+
+If you want to modify an existing Appsody stack navigate to the [stacks repo](https://github.com/appsody/stacks). Clone the stacks repo and navigate to the stack you want to modify:
+
+```bash
+git clone https://github.com/appsody/docs
```
Comment (helenmasters) [inline docs/stacks/create-or-modify.md]: Why do the instructions say navigate to the stacks repo and then clone the docs repo?

File: docs/stacks/create-or-modify.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    └── my-starter/
+        └── .appsody-config.yaml
+```
+
+## Image
+
+The image provides foundational capabilities that can be extended or replaced by the user application that is based on it. The image has mechanisms to control which aspects can and can not be overridden by the developer.
+
+### Project directory:
+The project folder should contain a production [Dockerfile](#Dockerfile) for your application and the project you are going to provide as a content provider.
```
Comment (helenmasters) [inline docs/stacks/create-or-modify.md]: provide as a content provider? can we lose one of the provides?

File: docs/stacks/create-or-modify.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+|   |   └── Dockerfile
+│   └── Dockerfile-stack
+└── templates/
+    └── my-starter/
+        └── .appsody-config.yaml
+```
+
+## Image
+
+The image provides foundational capabilities that can be extended or replaced by the user application that is based on it. The image has mechanisms to control which aspects can and can not be overridden by the developer.
```
Comment (helenmasters) [inline docs/stacks/create-or-modify.md]: can not -> cannot

File: docs/stacks/create-or-modify.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Modifying a stack
+
+If you want to modify an existing Appsody stack navigate to the [stacks repo](https://github.com/appsody/stacks). Clone the stacks repo and navigate to the stack you want to modify:
+
+```bash
+git clone https://github.com/appsody/docs
+cd docs/incubator/nodejs-express
+```
+
```
Comment (helenmasters) [inline docs/stacks/create-or-modify.md]: The stack structure might be better in another file or certainly further down here. What I'd like to see next at this point is an example stepping through creating or modifying a stack.

We have the following content which you could use to base an example on:

## Modifying the Java Spring Boot 2 stack

1. Clone the GitHub repository:

    ```bash
    git clone https://github.com/appsody/stacks
    ```

2. Package the starters:

    ```bash
    cd paks
    ./ci/packageStarters.sh .
    ```

3. Build the `java-spring-boot2` stack image:

    ```bash
    cd incubator/java-spring-boot2/stack
    docker build -t $(cat ../starters/default/.knapp-config.yaml | cut -d' ' -f2) -f Dockerfile-stack .
    ```

4. Test the stack by creating a project:

    ```bash
    mkdir /tmp/test-me
    cd /tmp/test-me
    appsody init java-spring-boot2
    ```

5. Run the application:

    ```bash
    appsody run
    ```

**Comment by mhamwala** [comment]
closing because changes addressed in this PR: https://github.com/appsody/docs/pull/11

**Pull request closed without merging**